packages/utils/src/lib/git.ts

```diff
@@ -17,3 +17,40 @@ export async function getLatestCommit() {
   });
   return log.latest;
 }
+
+export async function branchHasChanges(): Promise<boolean> {
+  return await git.status(['-s']).then(r => r.files.length > 0);
+}
+
+export async function guardAgainstDirtyRepo(): Promise<void> {
+  const isDirty = await branchHasChanges();
+  if (isDirty) {
+    throw new Error(`
+        Repository should be clean before we you can proceed.
+        Commit your local changes or stash them.
+      `);
+  }
+}
+
+export async function getCurrentBranchOrTag(): Promise<string> {
+  return (
+    (await git.branch().then(r => r.current)) ||
+    // @TODO replace with simple git
+    (await git.raw(['describe --tags --exact-match']).then(out => out.trim()))
+  );
+}
```

Comment:
What if there is no current branch or tag?

Reply:
No, mean what if the current commit isn't tagged. How do you handle this output?

![image](https://github.com/code-pushup/cli/assets/34691111/724a1623-88b8-49ed-87ed-7ac71ef94cc0)

Reply:
Throw?😇

I will implement a check for this case.

Reply:
I have no clue what i should do without the information. So I would go for the current implementation